### PR TITLE
Setup

### PR DESCRIPTION
**Pull request by jreadey**
Updates to make it more convenient to directly run hsds app.

### CODE REVIEW

File: hsds/app.py [hunk 1]
```diff
@@ -241,16 +249,25 @@ def main():
         print(f"unsupported log_level: {log_level_cfg}, using INFO instead")
         log_level = logging.INFO
 
-    print("set logging to:", log_level)
+    print("set logging to::", log_level)
```
Comment (mattjala) [inline hsds/app.py:252]: Typo?

File: README.md [hunk 2]
```diff
@@ -5,27 +5,27 @@
...
-To shut down the server, and the server was started with the --no-docker option, just control-C.
+Make sure you have Python 3 and Pip installed, then:
+
+1.  Run install: `$ ./build.sh` from source tree OR install from pypi: `$ pip install hsds`
+2.  Create a directory the server will use to store data, example: `$ mkdir ~/hsds_data`
+3.  Start server: `$ hsds --root_dir ~/hsds_data`
+4.  Run the test suite. In a separate terminal run:
+    -  Set user_name: `$ export USER_NAME=$USER`
+    -  Set user_password: `$ export USER_PASSWORD=$USER`
+    -  Set admin name: `$ export ADMIN_USER=$USER`
```
Comment (mattjala) [inline README.md:22]: `ADMIN_USER` -> `ADMIN_USERNAME`

**Comment by mattjala** [comment]
The unix socket tests and docker tests will work with the new build. Besides the `ADMIN_USER` -> `ADMIN_USERNAME` change in the README, LGTM.

**Review by mattjala — APPROVED** [review]

File: hsds/app.py [hunk 3]
```diff
@@ -329,6 +331,27 @@ def main():
...
+    else:
+        root_dir = config.get("root_dir")
+    if not root_dir:
+        # check that AWS_S3_GATEWAY or AZURE_CONNECTION_STRING is set
+        if not config.get("aws_s3_gateway") and not config.get("azure_connection_string"):
+            sys.exit("root_dir not set (and no S3 or Azure connection info)")
+    else:
+        if not os.path.isdir(root_dir):
+            sys.exit(f"directory: {root_dir} not found")
+        bucket_path = os.path.join(root_dir, bucket_name)
```
Comment (kandersolar) [inline hsds/app.py:351]: @jreadey I think there is an issue here.  The bucket name retrieved from the arg parser is in the form of a single-element list due to specifying [nargs=1](https://docs.python.org/3/library/argparse.html#nargs). This call to `os.path.join` then errors when passed a list.

**Comment by jreadey** [comment]
yes, no sure how the nargs got in there. 
I've checked in a fix on master.

**Pull request merged**